Add eslint-airbnb-config/legacy to project

jadengore opened the pull request:
This PR makes two major changes:
- Adds ESLint to the project, using Airbnb's legacy rules (slightly modified, see `.eslintrc`, all changes relevant to the project). I've added it as a `posttest` step to `npm test`, so no CircleCI configuration modification required. Running `npm run lint` lints the codebase in development.
    * This change also caught some unused dependencies, and they have been removed as well.
- Updates deployment test to work on forks. The test was failing for me because `airbrake/node-airbrake.git` was hard-coded as the repo, but since I am working on a fork this broke.

I'm willing to squash this branch and field some feedback, but I think this is at a state where it's ready for review.

phumpal commented on lib/airbrake.js:
```diff
@@ -138,61 +138,61 @@ Airbrake.prototype._sendRequest = function(err, cb) {
     }
 
-    if (undefined === body) {
+    if (typeof responseBody === 'undefined') {
```
This looks like a style win as well as being safer. Can you explain the difference between this and say

```
if (typeof responseBody ===  typeof undefined)
```

Which is better?  

Or did you use the `'undefined'` string because Lint fails with `typeof undefined`?

jadengore replied:
Normally, in JS land I believe it is enough to just allow falsy values to be evaluated in the conditional like so:
```js
if (!responseBody) {
  return callback(new Error('invalid body'));
}
```

However, I've been trying to maintain a similar style to the author's original check. The order of evaluation looked weird, and for the most part in JS land I think people use `typeof x === 'undefined'` so you don't have to evaluate the type twice.

phumpal commented on lib/airbrake.js:
```diff
@@ -336,36 +343,36 @@ Airbrake.prototype.trackDeployment = function(params, cb) {
   }, this.requestOptions);
 
-  var callback = this._callback(cb);
+  var requestCallback = this._callback(callback);
```
:+1: for renaming var.

phumpal commented on lib/airbrake.js:
```diff
@@ -50,59 +48,61 @@ function Airbrake() {
-  if (!(err instanceof Error)) {
-    err = new Error(err);
-  }
+  var error = (err instanceof Error) ? err : new Error(err);
```
While other node style guides suggest using multi-lines for ternaries the AirBNB guide prefers a [single line](https://github.com/airbnb/javascript#comparison--nested-ternaries) ternary.

:+1: for readability and being consistent.

jadengore replied:
@phumpal In this case, I think this is a good cleanup, especially when compared to the original conditional expression. It looks much cleaner in a ternary 😄

phumpal commented:
@jadengore Thanks for this. Overall, LGTM especially the ESLint coverage and style cleanups. Please seem my comments inline.

/cc @zefer @kyrylo @vmihailenco - Can I get a second set of eyes on this?

jadengore commented:
@phumpal This is actually a first in hopefully a series of changes I hope to help with. 

I am working up a draft for a proposed API modification, since I think even with the new style guide in place there are big improvements to be made (for example, all of the library's methods are attached to the prototype). Hope my changes get accepted!

kyrylo commented:
> This PR makes two major changes

Would it be possible to break up this PR in two smaller ones, then?

jadengore commented:
@kyrylo If I remove the deployment test fix, then my branch will fail CI.

phumpal commented:
@jadengore if that's the case. let's keep it as one. 

Can you squash such that the two major commits are separate changes? If that logical separation isn't possible then 1 commit should do.

Appreciate your work here! Thank you.

jadengore commented:
@phumpal I've squashed and updated into two commits.

@kyrylo @phumpal Thanks for the review! Excited to get this merged and have a cleaner codebase :smile:

phumpal commented:
@jadengore builds are now broken :( 

```
npm install
npm WARN peerDependencies The peer dependency eslint-plugin-import@^1.6.0 included from eslint-config-airbnb-base will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency 
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
npm WARN engine wreck@6.3.0: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.33","npm":"2.13.5"})
npm WARN engine joi@6.10.1: wanted: {"node":">=0.10.40","npm":">=2.0.0"} (current: {"node":"0.10.33","npm":"2.13.5"})
npm WARN engine deep-extend@0.4.1: wanted: {"node":">=0.12.0","iojs":">=1.0.0"} (current: {"node":"0.10.33","npm":"2.13.5"})
npm WARN engine hoek@2.16.3: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.33","npm":"2.13.5"})
npm WARN engine boom@2.10.1: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.33","npm":"2.13.5"})
npm WARN engine boom@2.10.1: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.33","npm":"2.13.5"})
npm WARN engine hoek@2.16.3: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.33","npm":"2.13.5"})
npm WARN engine cryptiles@2.0.5: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.33","npm":"2.13.5"})
npm WARN engine hoek@2.16.3: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.33","npm":"2.13.5"})
npm WARN engine topo@1.1.0: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.33","npm":"2.13.5"})
npm ERR! Linux 3.13.0-83-generic
npm ERR! argv "node" "/home/ubuntu/nvm/v0.10.33/bin/npm" "install"
npm ERR! node v0.10.33
npm ERR! npm  v2.13.5
npm ERR! code EPEERINVALID

npm ERR! peerinvalid The package eslint-plugin-import@1.5.0 does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer eslint-config-airbnb-base@1.0.4 wants eslint-plugin-import@^1.6.0

npm ERR! Please include the following file with any support request:
npm ERR!     /home/ubuntu/node-airbrake/npm-debug.log

npm install returned exit code 1

Action failed: npm install
```

jadengore commented:
Looks like `eslint-plugin-import` is an unmet dependency and CircleCi is failing it. I am opening a PR to address!
